prestodb/transaction.py

```diff
@@ -0,0 +1,81 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import logging
+
+from prestodb import constants
+import prestodb.exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+class Transaction(object):
+    def __init__(self, request):
+        self._request = request
+        self._id = 'NONE'
+
+    @property
+    def id(self):
+        return self._id
+
+    def begin(self):
+        response = self._request.post('START TRANSACTION')
+        if not response.ok:
+            raise prestodb.exceptions.DatabaseError(
+                'failed to start transaction: {}'.format(response.status_code))
+        transaction_id = response.headers.get(
+            constants.HEADER_STARTED_TRANSACTION
+        )
+        if transaction_id and transaction_id != 'NONE':
+            self._id = (
+                response.headers[constants.HEADER_STARTED_TRANSACTION]
+            )
+        status = self._request.process(response)
+        while status.next_uri:
```

Comment:
hmmmmm... is this some how duplicated with the logic in client.py?

Reply:
Yes the logic is very similar and it bothered me too. However it was really simpler that way. I think this could be addressed by a follow-up refactoring.